File: Python 3/Data_Structures/Arrays/arrayMethods.py
# Arrays : Organize items sequentially

strings = ['a','b','c','d']
# 4 per each item = 16 bytes of storage used

print(strings[2])

# Push: Add item at the end 
strings.append('e')      # O(1)

# Pop: Delete item at the end
strings.pop() 
strings.pop()            # O(1)

# Addfirstelement: Add element at the beggining (position 0)
strings.insert(0,'x')    #O(n)

# Splice: Divide the array
# Insert: its used to insert in the desire position
strings.insert(2,'alien')   #O(n)

print(strings)

File: JavaScript/data_Structures/Arrays/arrayMethods.js
// Arrays : Organize items sequentially 

const strings = ['a','b','c','d']
// 4 shells per item = 16 bytes of storage

strings[2]

// Push: Adds at the end of the array
strings.push('e');      // O(1)
console.log(strings);  

// Pop: Remove last item
strings.pop();         // O(1)
console.log(strings);  

// Unshift: Adds at the begginig of the array
strings.unshift('x');   // O(n)
console.log(strings);


/** Splice: Can divide the array into parts
/* Can also be used to insert in the
 * desire position */
strings.splice(2,0,'alien');   // O(n)
console.log(strings);

